model/openapi.yml

```diff
@@ -911,4 +1002,99 @@ paths:
           $ref: '#/components/responses/throttle'
         '503':
           $ref: '#/components/responses/unavailable'
-# TODO: Include the file-upload endpoints?
+  /api/fleet/uploads:
+    post:
+      summary: Initiate a file upload process
+      description: ""
+      security:
+        - agentApiKey: []
+      requestBody:
+        content:
+          application/json:
+            schema:
+              $ref: '#/components/schemas/uploadBeginRequest'
+      responses:
+        '200':
+          description: Information about the upload procedure
+          content:
+            application/json:
+              schema:
+                $ref: '#/components/schemas/uploadBeginResponse'
+        '400':
+          $ref: '#/components/responses/badRequest'
+  /api/fleet/uploads/{id}/{chunkNum}:
+    put:
+      summary: Upload a section of file data
+      description: "Upload portions of the intended file in a piecewise fashion. This route is idempotent in that any successive calls will overwrite previous ones for the same URL parameters. Chunks may be uploaded in any order, and may be uploaded in parallel. The body is the raw contents of the file at the given position matching the chunk number. The body must be the exact chunk size returned from the upload initiation response. All chunks must be this size except for the final one, which is naturally the file remainder."
+      security:
+        - agentApiKey: []
+      parameters:
+        - name: id
+          in: path
+          description: The upload_id as returned in the Upload initiation response
+          required: true
+          schema:
+            type: string
+            format: uuid
+            example: ecb30383-6dd1-4b1d-bed0-2386b4e5df51
+        - name: chunkNum
+          in: path
+          description: the positional index of the chunk within the file. The first chunk is 0, the next 1, etc.
+          required: true
+          schema:
+            type: integer
+            example: 3
+        - name: X-Chunk-SHA2
+          in: header
+          required: true
+          description: the SHA256 hash of the body contents for this request
+          schema:
+            type: string
+            example: 0c4a81b85a6b7ff00bde6c32e1e8be33b4b793b3b7b5cb03db93f77f7c9374d1
+      requestBody:
+        description: The chunk contents as bytes
+        required: true
+        content:
+            "*/*":
```

Comment:
do we want to use `application/octet-stream` instead?

Reply:
to be fair, `*/*` is the most accurate. The content-type is not checked, and the actual content could be plain text, could be binary, could be anything.

A client could send

`Content-Type: text/csv` and send just text in the body and that would work.